source/common/router/router.h

```diff
@@ -434,6 +435,9 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   bool is_retry_ : 1;
   bool include_attempt_count_ : 1;
   bool attempting_internal_redirect_with_complete_stream_ : 1;
+  // Tracks whether we deferred a per try timeout because the downstream request
+  // had not been completed yet.
+  bool pending_per_try_timeout_ : 1;
```

Comment:
let's put this on the UpstreamRequest instead of on the parent, that will hold up better once we have multiple upstream requests